platform/include/tfm_hal_its.h

```diff
@@ -73,6 +78,121 @@ extern ARM_DRIVER_FLASH TFM_HAL_ITS_FLASH_DRIVER;
 enum tfm_hal_status_t
 tfm_hal_its_fs_info(struct tfm_hal_its_fs_info_t *fs_info);
 
+#ifdef TFM_ITS_ENCRYPTED
+
+/**
+ * \brief Generate an encryption nonce
+ *
+ * \details The nonce has to be unique for every encrypted file. A pseudorandom
+ *          number is highly reccomended but not enforced. Each platform need
+ *          to enforce their own strategy to differentiate the nonces.
+ *
+ * \param [out] nonce           Pointer to the nonce
+ * \param [in]  nonce_size      Size of the nonce in bytes
+ *
+ * \retval TFM_HAL_SUCCESS             The operation completed successfully
+ * \retval TFM_HAL_ERROR_INVALID_INPUT Invalid argument
+ * \retval TFM_HAL_ERROR_GENERIC       Failed to fill the nonce seed because of
+ *                                     an internal error
+ */
+enum tfm_hal_status_t tfm_hal_its_aead_generate_nonce(uint8_t *nonce,
+                                                      size_t nonce_size);
+
+/**
+ * \brief Set the derivation label for the AEAD operation.
+ *
+ * \param [in] ctx               AEAD context for ITS object
+ * \param [in] deriv_label       Pointer to the derivation label
+ * \param [in] derive_label_size Size of the derivation label in bytes
+ *
+ * \retval TFM_HAL_SUCCESS             The operation completed successfully
+ * \retval TFM_HAL_ERROR_INVALID_INPUT Invalid argument
+ */
+enum tfm_hal_status_t tfm_hal_its_aead_set_deriv_label(
+                                         struct tfm_hal_its_auth_crypt_ctx *ctx,
+                                         uint8_t *deriv_label,
+                                         size_t deriv_label_size);
+
+/**
+ * \brief Set the nonce for the AEAD operation.
+ *
+ * \param [in] ctx        AEAD context for ITS object
+ * \param [in] nonce      Pointer to the nonce
+ * \param [in] nonce_size Size of the nonce in bytes
+ *
+ * \retval TFM_HAL_SUCCESS             The operation completed successfully
+ * \retval TFM_HAL_ERROR_INVALID_INPUT Invalid argument
+ */
+enum tfm_hal_status_t tfm_hal_its_aead_set_nonce(
+                                         struct tfm_hal_its_auth_crypt_ctx *ctx,
+                                         uint8_t *nonce,
+                                         size_t nonce_size);
+
+/**
+ * \brief Set the additional authenticated data for the AEAD operation.
```

Comment:
```suggestion
 * \brief Set the pointer to the additional authenticated data for the AEAD operation.
```

Let's make it more clear that no copying needs to be done here.